crypto/cipher_extra/e_tls.c

```diff
@@ -45,19 +45,12 @@ typedef struct {
 OPENSSL_STATIC_ASSERT(EVP_MAX_MD_SIZE < 256,
                       mac_key_len_does_not_fit_in_uint8_t)
 
-OPENSSL_STATIC_ASSERT(sizeof(((EVP_AEAD_CTX *)NULL)->state) >=
-                          sizeof(AEAD_TLS_CTX),
-                      AEAD_state_is_too_small)
-#if defined(__GNUC__) || defined(__clang__)
-OPENSSL_STATIC_ASSERT(alignof(union evp_aead_ctx_st_state) >=
-                          alignof(AEAD_TLS_CTX),
-                      AEAD_state_has_insufficient_alignment)
-#endif
-
 static void aead_tls_cleanup(EVP_AEAD_CTX *ctx) {
-  AEAD_TLS_CTX *tls_ctx = (AEAD_TLS_CTX *)&ctx->state;
+  AEAD_TLS_CTX *tls_ctx = (AEAD_TLS_CTX *)ctx->state.ptr;
```

Comment:
You deleted the static asserts. What about the other static asserts in the code?
```
grep -r "evp_aead_ctx_st_state" ./                                                                                                                                                      
.//crypto/fipsmodule/cipher/e_aesccm.c:OPENSSL_STATIC_ASSERT(alignof(union evp_aead_ctx_st_state) >=                                                                                            
.//crypto/fipsmodule/cipher/e_aes.c:OPENSSL_STATIC_ASSERT(alignof(union evp_aead_ctx_st_state) >=                                                                                               
.//crypto/fipsmodule/cipher/e_aes.c:OPENSSL_STATIC_ASSERT(alignof(union evp_aead_ctx_st_state) >=                                                                                               
.//crypto/fipsmodule/cipher/e_aes.c:OPENSSL_STATIC_ASSERT(alignof(union evp_aead_ctx_st_state) >=                                                                                               
Binary file .//crypto/fipsmodule/rand/perf_drbg_changes matches                                                                                                                                 
.//crypto/cipher_extra/e_chacha20poly1305.c:OPENSSL_STATIC_ASSERT(alignof(union evp_aead_ctx_st_state) >=                                                                                       
.//crypto/cipher_extra/e_aesctrhmac.c:OPENSSL_STATIC_ASSERT(alignof(union evp_aead_ctx_st_state) >=                                                                                             
.//crypto/cipher_extra/e_aesgcmsiv.c:OPENSSL_STATIC_ASSERT(alignof(union evp_aead_ctx_st_state) >= 8,                                                                                           
.//crypto/cipher_extra/e_aesgcmsiv.c:OPENSSL_STATIC_ASSERT(alignof(union evp_aead_ctx_st_state) >=
```
I haven't looked into whether these can now fail.